### PR TITLE
Added manual cooling and VWZ binding

### PR DESCRIPTION
**Pull request by kolibrie-eric**
Added manual cooling option as offered by the 700 when you have a heat pump installed. Added VWZ (heatpump controller) items to the hmu configuration file. Need help with translation though

### CODE REVIEW

**Comment by kjoglum** [comment]
@kolibrie-eric 
I am testing your modified config files for my Vaillant heating setup: VWL 85/2, VRC 700, VWZ. 

Although, I do not get any reasonable values for LiveMonitor. I am interested in getting the "Current Power Consumption", so I have been trying:

```
pi@raspberrypi:~$ ebusctl write -c hmu ReadLiveMonitor 003224
done

pi@raspberrypi:~$ ebusctl read -v LiveMonitor
hmu LiveMonitor subid=24;value=0c 00
```

**Comment by kolibrie-eric** [comment]
@kjoglum, 

The command is okay. It gets you the current power consumption when you read the LiveMonitor value as you did. The actual value 0c00 is a UIN and needs interpretation. Here is a piece of Javascript code to help you decode its value (UIN):

function UIN(lbyte,hbyte)
{
    if (hbyte.length!=2) return null;
    if (lbyte.length!=2) return null;
    return parseInt(hbyte+lbyte,16)
}

The answer is 12kW.

**Comment by kjoglum** [comment]
@kolibrie-eric 

Thanks for sharing! 

Do you also have some sort of automatic repeat sequence to trigger LiveMonitor, e.g. MQTT? I am successfully able to trigger LiveMonitor manually from ebusd using `mosquitto_pub -m '003224' -t 'ebusd/hmu/ReadLiveMonitor/set'`, but uncertain how to repeat automatically (similar to `-m '?1'` as used for poll priority for /get values).

**Comment by kolibrie-eric** [comment]
@kjoglum 

I created the repeat sequence in node red where you can poll the values and receive updates through the MQTT interface. Basically, you send the ebusctl message to the TCP/IP port 8888, localhost.

Good luck!

**Comment by kjoglum** [comment]
@kolibrie-eric 

Managed to do the same with a scripted Time cron rule using the MQTT binding / actions for OpenHab. Works perfectly, thanks!

**Comment by lfrlucas** [comment]
Hi, when can we expect to see this branch merged? Amazing feature, the ability to get VWZ live monitor settings.

**Comment by john30** [comment]
can you post your scan result please? I'd like to make sure that this does not influence different versions of the 700

**Comment by lfrlucas** [comment]
@john30 do you mean this?
08;Vaillant;HMU01;0304;8802;21;17;47;0010019765;1610;005914;N5
15;Vaillant;70000;0510;6403;21;18;31;0020171318;0082;008286;N9
76;Vaillant;VWZIO;0111;0103;21;18;25;0020117049;0082;007307;N0
e8;Vaillant;FMU00;0202;6502;21;17;47;0010019765;1610;005914;N5
note that I have created local configurations 76.vwzio and e8.fmu

File: ebusd-2.1.x/de/vaillant/08.hmu.csv [hunk 1]
```diff
@@ -1,6 +1,37 @@
 # type (r[1-9];w;u),circuit,name,[comment],[QQ],ZZ,PBSB,[ID],field1,part (m/s),datatypes/templates,divider/values,unit,comment,field2,part (m/s),datatypes/templates,divider/values,unit,comment,field3,part (m/s),datatypes/templates,divider/values,unit,comment,field4,part (m/s),datatypes/templates,divider/values,unit,comment
+*r,,,,,,"B507",09,,,,,,
+#r,,Unknown,,,,,,,m,temp1,,,
+*w,,,,,,"B507",09,,,,,,
+#w,,Unknown,,,,,,,m,temp1,,,
```
Comment (john30) [inline ebusd-2.1.x/de/vaillant/08.hmu.csv:5]: unused, please remove (same in "en" file)

**Review by john30 — CHANGES_REQUESTED** [review]
there are more changes to be done for ease of use.
I would need to get some samples as hex. can someone provide these?

**Comment by kolibrie-eric** [comment]
Hi John, Here are some examples of the output (I removed the status translation from the ebusd configuration files in my setup as I have a multiple language user interface) In 08.hmu.csv  VWZ Live monitor values: Main status:[update info] sent MS cmd: 3108b503020003 / 0af401ffffffffffffffff[update notice] sent read hmu LiveMonitorMain: 244;Heating Modulation. Datatype is D1B[update info] sent MS cmd: 3108b51a0405003225 / 0a00081b00000000000000[update notice] sent read hmu LiveMonitor QQ=31: 25;00 00 Power consumption. Datatype is UIN[update info] sent MS cmd: 3108b51a0405003224 / 0a00083500000000000000[update notice] sent read hmu LiveMonitor QQ=31: 24;00 00 Air intake temperature. Datatype is D2C[update info] sent MS cmd: 3108b51a0405003226 / 0a000808df000000000000[update notice] sent read hmu LiveMonitor QQ=31: 26;df 00 Desired supply temperature. Datatype is D2C[update info] sent MS cmd: 3108b51a040500321f / 0a000805f0000000000000[update notice] sent read hmu LiveMonitor QQ=31: 1f;f0 00 Actual supply temperature. Datatype is D2C[update info] sent MS cmd: 3108b51a0405003220 / 0a0008081b010000000000[update notice] sent read hmu LiveMonitor QQ=31: 20;1b 01 Power generated. Datatype is UIN[update info] sent MS cmd: 3108b51a0405003223 / 0a00083500000000000000[update notice] sent read hmu LiveMonitor QQ=31: 23;00 00 VWZ test menu’s: ReadEEVPosition[update info] sent MS cmd: 3108b51405051503ffff / 0415002600[update notice] sent read hmu EEVPosition QQ=31: 38 ReadCompressorIntakeTemp[update info] sent MS cmd: 3108b51405053803ffff / 0438008300[update notice] sent read hmu CompressorIntakeTemp QQ=31: 13.1 Additional thermostat items (tested for 15.700.csv and 15.b7v00.csv (wireless 700)): [update info] sent MS cmd: 3115b524080201000006000000 / 020000[update notice] sent write b7v00 ManualCooling QQ=31: 0 There are many more items but they all work similar to the menu items above Let me know if you need more information Eric Van: JohnVerzonden: zaterdag 20 februari 2021 10:35Aan: john30/ebusd-configurationCC: kolibrie-eric; MentionOnderwerp: Re: [john30/ebusd-configuration] Added manual cooling and VWZ binding (#160) @john30 requested changes on this pull request.there are more changes to be done for ease of use.I would need to get some samples as hex. can someone provide these?In ebusd-2.1.x/de/vaillant/08.hmu.csv:> +*r,,,,,,"B507",09,,,,,,+#r,,Unknown,,,,,,,m,temp1,,,+*w,,,,,,"B507",09,,,,,,+#w,,Unknown,,,,,,,m,temp1,,,unused, please remove (same in "en" file)—You are receiving this because you were mentioned.Reply to this email directly, view it on GitHub, or unsubscribe.

**Comment by john30** [comment]
can't read your last comment. please reformat

File: ebusd-2.1.x/de/vaillant/08.hmu.csv (hunk 1 above)
Comment (kolibrie-eric) [reply ebusd-2.1.x/de/vaillant/08.hmu.csv:5]: Hi John,

Here are some examples of the output (I removed the status translation from the ebusd configuration files in my setup as I have a multiple language user interface)

In 08.hmu.csv

VWZ Live monitor values:

Main status:
[update info] sent MS cmd: 3108b503020003 / 0af401ffffffffffffffff
[update notice] sent read hmu LiveMonitorMain: 244;Heating

Modulation. Datatype is D1B
[update info] sent MS cmd: 3108b51a0405003225 / 0a00081b00000000000000
[update notice] sent read hmu LiveMonitor QQ=31: 25;00 00

Power consumption. Datatype is UIN
[update info] sent MS cmd: 3108b51a0405003224 / 0a00083500000000000000
[update notice] sent read hmu LiveMonitor QQ=31: 24;00 00

Air intake temperature. Datatype is D2C
[update info] sent MS cmd: 3108b51a0405003226 / 0a000808df000000000000
[update notice] sent read hmu LiveMonitor QQ=31: 26;df 00

Desired supply temperature. Datatype is D2C
[update info] sent MS cmd: 3108b51a040500321f / 0a000805f0000000000000
[update notice] sent read hmu LiveMonitor QQ=31: 1f;f0 00
 
Actual supply temperature. Datatype is D2C
[update info] sent MS cmd: 3108b51a0405003220 / 0a0008081b010000000000
[update notice] sent read hmu LiveMonitor QQ=31: 20;1b 01

Power generated. Datatype is UIN
[update info] sent MS cmd: 3108b51a0405003223 / 0a00083500000000000000
[update notice] sent read hmu LiveMonitor QQ=31: 23;00 00

VWZ test menu’s:

ReadEEVPosition
[update info] sent MS cmd: 3108b51405051503ffff / 0415002600
[update notice] sent read hmu EEVPosition QQ=31: 38

ReadCompressorIntakeTemp
[update info] sent MS cmd: 3108b51405053803ffff / 0438008300
[update notice] sent read hmu CompressorIntakeTemp QQ=31: 13.1

Additional thermostat items (tested for 15.700.csv and 15.b7v00.csv (wireless 700)):
[update info] sent MS cmd: 3115b524080201000006000000 / 020000
[update notice] sent write b7v00 ManualCooling QQ=31: 0

There are many more items but they all work similar to the menu items above

Let me know if you need more information
Eric

**Comment by stefanheijnen** [comment]
How is this PR progressing? Also I see some errors in your https://github.com/kolibrie-eric/ebusd-configuration/blob/master/ebusd-2.1.x/en/vaillant/08.hmu.csv file @kolibrie-eric 

Line 3, 19, 36 etc. Looks like an error with merging or so?

I'm very interested in the data from the live monitor. I install and service heatpumps every day. Put programming is not in my skill set unfortunately.

**Comment by LukasGrebe** [comment]
> can't read your last comment. please reformat

I've take that liberty, seeing @kolibrie-eric hasn't responded:

> Hi John, Here are some examples of the output (_I removed the status translation from the ebusd configuration files in my setup as I have a multiple language user interface_)
> 
> In 08.hmu.csv  VWZ Live monitor values: Main status:
> 
> [update info] sent MS cmd: 3108b503020003 / 0af401ffffffffffffffff
> [update notice] sent read hmu LiveMonitorMain: 244;Heating Modulation. Datatype is D1B
> [update info] sent MS cmd: 3108b51a0405003225 / 0a00081b00000000000000
> [update notice] sent read hmu LiveMonitor QQ=31: 25;00 00 Power consumption. Datatype is UIN
> [update info] sent MS cmd: 3108b51a0405003224 / 0a00083500000000000000
> [update notice] sent read hmu LiveMonitor QQ=31: 24;00 00 Air intake temperature. Datatype is D2C
> [update info] sent MS cmd: 3108b51a0405003226 / 0a000808df000000000000
> [update notice] sent read hmu LiveMonitor QQ=31: 26;df 00 Desired supply temperature. Datatype is D2C
> [update info] sent MS cmd: 3108b51a040500321f / 0a000805f0000000000000
> [update notice] sent read hmu LiveMonitor QQ=31: 1f;f0 00 Actual supply temperature. Datatype is D2C
> [update info] sent MS cmd: 3108b51a0405003220 / 0a0008081b010000000000
> [update notice] sent read hmu LiveMonitor QQ=31: 20;1b 01 Power generated. Datatype is UIN
> [update info] sent MS cmd: 3108b51a0405003223 / 0a00083500000000000000
> [update notice] sent read hmu LiveMonitor QQ=31: 23;00 00 VWZ test menu’s: ReadEEVPosition
> [update info] sent MS cmd: 3108b51405051503ffff / 0415002600
> [update notice] sent read hmu EEVPosition QQ=31: 38 ReadCompressorIntakeTemp
> [update info] sent MS cmd: 3108b51405053803ffff / 0438008300
> [update notice] sent read hmu CompressorIntakeTemp QQ=31: 13.1 Additional thermostat items (tested for 15.700.csv and 15.b7v00.csv (wireless 700)): [update info] sent MS cmd: 3115b524080201000006000000 / 020000
> [update notice] sent write b7v00 ManualCooling QQ=31: 0
> 
> There are many more items but they all work similar to the menu items above
> 
> Let me know if you need more information
> 
> Eric

**Comment by karusell** [comment]
> 

Could you provide these local files for 76.vwzio and e8.fmu ? I would like to get also the 76.vzwio data. 

Thank you.

**Comment by lfrlucas** [comment]
@karusell I also need those ones. I solved by creating 76.vwzio and e8.fmu from existing 08.hmu.csv. It seems to work for some fields.

**Comment by karusell** [comment]
If we can find out the config for VWZIO it is for both technical components "Vaillant VWZ AI" and "Vaillant VWZ MEH 97/6" because they are identically construced control units.

@lfrlucas could you provide me your files? thank you!

**Comment by lfrlucas** [comment]
my files are a simple copy of 08.hmu.csv

**Comment by john30** [comment]
reworked+merged in 6577542

**Comment by john30** [comment]
please check the [next](https://ebus.github.io/next/) generated csvs with ebusd 24.1 (or current source) as described there or by using a local clone of https://github.com/eBUS/ebus.github.io and using the "next" folder in there

**Comment by bernd-wiswedel** [comment]
It seems this PR removes `WaterThroughput` from the `08.hmu.csv`. The corresponding [commit](https://github.com/john30/ebusd-configuration/commit/6577542f0188cbe081cf0b926f7541d9f9e63d1c#diff-469cd14907906c02349f5d5217ade258d9a476bac864029119b06e2e0a3c9162L51) (22nd Dec 24) itself isn't shown to be part of this PR but the commit message suggests there is a link. 

`WaterThroughput` is gone when switching to ebusd 24.1, as also noted by others on [this ebusd discussion](https://github.com/john30/ebusd/discussions/1476).

**Comment by john30** [comment]
> It seems this PR removes `WaterThroughput` from the `08.hmu.csv`. Here is the corresponding line in commit [6577542](https://github.com/john30/ebusd-configuration/commit/6577542f0188cbe081cf0b926f7541d9f9e63d1c#diff-469cd14907906c02349f5d5217ade258d9a476bac864029119b06e2e0a3c9162L51).
> 
> `WaterThroughput` is gone when switching to ebusd 24.1, as also noted by others on [this ebusd discussion](https://github.com/john30/ebusd/discussions/1476).

no, it was moved+renamed here:
https://github.com/john30/ebusd-configuration/blob/master/src/vaillant/08.hmu.tsp#L855-L867

**Comment by bernd-wiswedel** [comment]
(thanks)

I saw it but couldn't make sense out of it. I haven't found how to enable them (as per [comment](https://github.com/john30/ebusd-configuration/blob/master/src/vaillant/08.hmu.tsp#L791C6-L791C99)):

> // Test menus on VWZ. EnableTest message needs to be sent before each of the read messages work.

```
read -c hmu EnableTestWaterThroughput
ERR: element not found

read -c hmu TestWaterThroughput 
ERR: element not found

read -c hmu SourceTempOutput
9.81
```
(The latter just to confirm my own sanity)

I assume `EnableTestWaterThroughput` needs to be written to but what (blank value didn't do it).

**Comment by DRUROB** [comment]
Hi!
When i followed your link this this thread, i think i get what your up to...
But my adapter 24.1 is configured to get the files from the server  but it doesnt download or replace the files with the new *.tsp Version. 
I think they are for testing purposes, aren´t they? The name "test.....(anything)" provides it ? ;-)
But why is the parameter removed from the stable 15.700.csv version? Are there issues with it? 
The parameter waterthroughput is an important one when you want to see if there are to many circuits closed while heating.
Can you re-add it please?

**Comment by john30** [comment]
> > // Test menus on VWZ. EnableTest message needs to be sent before each of the read messages work.
> 
> ```
> read -c hmu EnableTestWaterThroughput
> ERR: element not found

you need to issue a write, not a read. this tells the receiver to allow reads afterwards

**Pull request closed without merging**